src/audio/tone-ipc4.c

```diff
@@ -0,0 +1,632 @@
+	 * echo reference, the number of input pins will be non-zero
+	 */
+	if (mod_config->nb_input_pins > 0) {
+		dev->direction = SOF_IPC_STREAM_CAPTURE;
+		cd->mode = TONE_MODE_SILENCE;
+
+	} else {
+		dev->direction = SOF_IPC_STREAM_PLAYBACK;
+		cd->mode = TONE_MODE_TONEGEN;
+	}
+
+	dev->direction_set = true;
+
+	/* Reset tone generator and set channels volumes to default */
+	for (i = 0; i < PLATFORM_MAX_CHANNELS; i++)
+		tonegen_reset(&cd->sg[i]);
+
+	return 0;
+}
+
+static int tone_free(struct processing_module *mod)
+{
+	struct comp_data *cd = module_get_private_data(mod);
+
+	comp_info(mod->dev, "tone_free()");
+
+	rfree(cd);
+	return 0;
+}
+
+/* set component audio stream parameters */
+static int tone_params(struct processing_module *mod)
+{
+	struct comp_data *cd = module_get_private_data(mod);
+	struct comp_dev *dev = mod->dev;
+	struct comp_buffer *sinkb;
+	enum sof_ipc_frame frame_fmt, valid_fmt;
+
+	sinkb = comp_dev_get_first_data_consumer(dev);
+	if (!sinkb) {
+		comp_err(dev, "sink buffer");
+		return -ENOTCONN;
+	}
+
+	audio_stream_fmt_conversion(mod->priv.cfg.base_cfg.audio_fmt.depth,
+				    mod->priv.cfg.base_cfg.audio_fmt.valid_bit_depth,
+				    &frame_fmt, &valid_fmt,
+				    mod->priv.cfg.base_cfg.audio_fmt.s_type);
+	cd->rate = mod->priv.cfg.base_cfg.audio_fmt.sampling_frequency;
+
+	comp_info(dev, "tone_params(), frame_fmt = %u", frame_fmt);
+
+	/* Tone supports only S32_LE PCM format atm */
+	if (frame_fmt != SOF_IPC_FRAME_S32_LE) {
+		comp_err(dev, "tone_params(), unsupported frame_fmt = %u", frame_fmt);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+/* copy and process stream data from source to sink buffers */
+static int tone_process(struct processing_module *mod,
+			struct sof_source **sources, int num_of_sources,
+			struct sof_sink **sinks, int num_of_sinks)
+{
+	struct comp_data *cd = module_get_private_data(mod);
+
+	if (num_of_sources > 0)
+		return cd->tone_func(mod, sinks[0], sources[0]);
+
+	return cd->tone_func(mod, sinks[0], NULL);
+}
+
+static int tone_prepare(struct processing_module *mod, struct sof_source **sources,
+			int num_of_sources, struct sof_sink **sinks, int num_of_sinks)
+{
+	struct comp_data *cd = module_get_private_data(mod);
+	struct comp_dev *dev = mod->dev;
+	int32_t f;
+	int32_t a;
+	int ret;
+	int i;
+
+	comp_info(dev, "tone_prepare()");
+
+	ret = tone_params(mod);
+	if (ret < 0)
+		return ret;
+
+	cd->channels = mod->priv.cfg.base_cfg.audio_fmt.channels_count;
+	comp_info(dev, "tone_prepare(), cd->channels = %u, cd->rate = %u",
+		  cd->channels, cd->rate);
+
+	for (i = 0; i < cd->channels; i++) {
+		f = tonegen_get_f(&cd->sg[i]);
+		a = tonegen_get_a(&cd->sg[i]);
+		if (tonegen_init(&cd->sg[i], cd->rate, f, a) < 0)
+			return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int tone_reset(struct processing_module *mod)
+{
+	struct comp_data *cd = module_get_private_data(mod);
+	int i;
+
+	comp_info(mod->dev, "tone_reset()");
```

Comment:
please check and remove all function names from logs, also please try to reduce the number of `*_info()` level logs, e.g. this one doesn't seem to be really important

Reply:
@ranj063 not sure what this consistency brings us. If GitHub could identify that this is a copy with minor changes from tone.c and would display only a difference between the files, then yes, reducing the difference helps reviewing, but GitHub displays it as a new file, so it isn't really that helpful IMHO